### PR TITLE
gentoo-common: PYTHON_TARGETS generation improvements

### PR DESCRIPTION
**Pull request by mgorny**
Teach the code to combine subsequent Python versions, i.e. instead of:

    PYTHON_COMPAT=( python3_{8,9,10} )

output:

    PYTHON_COMPAT=( python3_{8..10} )

While at it, simplify the code some.

CC @msva, @gentoo/python

### CODE REVIEW

**Review by msva — APPROVED** [review]
Btw, I thought of eval too, but I decided that implicit assignment would be "less-magical".

**Review by msva — APPROVED** [review]

File: plugin/gentoo-common.vim [hunk 1]
```diff
@@ -73,7 +73,6 @@ fun! GentooGetPythonTargets()
                 let l:m = l:py->matchstr("^python3.*")->matchstr("\\d*$")
                 if !empty(l:m)
                     eval l:py3s->add(l:m)
-                    continue
```
Comment (msva) [inline plugin/gentoo-common.vim:76]: Initially, it was for purpose, but I already forgot the reason.
IIRC it's an artefact of old PRE-merged revisions where there were another branch of code.
And now it is no need for it anymore.

Although, it would anyway be good to test if function would work as expected even on edge-cases after removing this.

**Review by msva — APPROVED** [review]
